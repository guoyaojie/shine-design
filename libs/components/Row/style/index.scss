/**
 * @Style 栅格系统 - Row
 * @Author 瞿龙俊 - qulongjun@shine.design
 * @Date 2018/8/31 下午1:59
 */

/**
 * 核心样式库
 */
@import "../../../sass/mixins/meida.scss";

/**
 * 依赖样式库
 */
@import "~bootstrap/scss/bootstrap-grid";

/**
  独立样式
 */
@import "grid.scss";