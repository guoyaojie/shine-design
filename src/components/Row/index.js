/**
 * @Component Row
 * @Type 布局组件
 * @Author 瞿龙俊 - qulongjun@shine.design
 * @Date 2019-03-16 02:02
 */
import React, {PureComponent} from 'react';
import _ from 'lodash';
import classNames from 'classnames';
import * as PropTypes from 'prop-types';
import {classPrefix} from '../../_variables';
import './style/index.scss';

/** 支持布局类型 */
const TYPES = ['flex'];
/** 交叉轴对齐方式 */
const ALIGN = ['top', 'middle', 'bootom'];
/** 主轴对齐方式 */
const JUSTIFY = ['start', 'end', 'center', 'space-around', 'space-between'];

class Row extends PureComponent {

  render() {
    const {
      justify,
      align,
      className,
      attributes,
      children,
    } = this.props;

    /** 计算样式 */
    const classes = classNames(
      `${classPrefix}-row`,
      `${classPrefix}-row--root`,
      {
        [`${classPrefix}-row--justify-${justify}`]: !_.isUndefined(justify),
        [`${classPrefix}-row--align-${align}`]: !_.isUndefined(align),
      },
      className,
    );

    return (
      <div
        className={classes}
        {...attributes}
      >
        {children}
      </div>
    );
  }
}

Row.propTypes = {
  /** 布局类型，目前仅支持 flex */
  type: PropTypes.oneOf(TYPES),
  /** 水平对齐方式 */
  justify: PropTypes.oneOf(JUSTIFY),
  /** 垂直对齐方式 */
  align: PropTypes.oneOf(ALIGN),
  /** 用户自定义修饰符 */
  className: PropTypes.string,
  /** 用户自定义属性 */
  attributes: PropTypes.object,
};

Row.defaultProps = {
  type: 'flex',
  justify: 'start',
  align: 'top',
  className: '',
  attributes: {},
};

export default Row;


