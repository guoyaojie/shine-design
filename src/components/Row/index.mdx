---
name: My Document
route: /custom-route
menu: Documents
---

# Hello world

Hello, I'm a mdx file!
