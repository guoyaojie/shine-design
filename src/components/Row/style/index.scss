/**
 * Row
 * @Author 瞿龙俊 - qulongjun@shine.design
 * @Date 2019-03-16 08:51
 */
@import "~style/variables";

@import "./mixins.scss";
@import "./variables";

@include shine--build-row($justify, $align, $classPrefix);
