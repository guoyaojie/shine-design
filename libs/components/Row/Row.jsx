/**
 * @Component 栅格系统 - Row
 * @Type Basic
 * @Author 瞿龙俊 - qulongjun@shine.design
 * @Date 2018/8/31 下午1:58
 */

// 核心依赖
import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';

// 第三方依赖库
import classNames from 'classnames'
import _ from 'lodash';

// 组件依赖

// 样式
import './style';

export default class extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        mode: PropTypes.oneOf(
            ['general', 'desktop', 'desktop-and-tablet', 'tablet', 'tablet-and-mobile', 'mobile']
        ).isRequired,
        direction: PropTypes.oneOf(
            ['ver', 'hor']
        ).isRequired,
        justify: PropTypes.oneOf(
            ['flex-start', 'flex-end', 'center', 'space-between', 'space-around']
        ),
        align: PropTypes.oneOf(
            ['flex-start', 'flex-end', 'center', 'baseline', 'stretch']
        ),
        alignContent: PropTypes.oneOf(
            ['flex-start', 'flex-end', 'center', 'space-between', 'space-around', 'stretch']
        ),
        wrap: PropTypes.oneOf(
            ['nowrap', 'wrap-reverse', 'wrap']
        )
    };

    static defaultProps = {
        mode: 'general',
        direction: 'hor'
    };

    render() {
        const me = this;
        const {mode, direction, justify, wrap, align, alignContent, className, children} = me.props;

        return (
            <Fragment>
                <div className={
                    classNames('m-grid', 'row', {
                            [`m-grid--${mode}`]: mode !== 'general',
                            [`m-grid--${direction}${mode !== 'general' ? '-' + mode : ''}`]: true,
                            [`m-grid--${justify}`]: _.isString(justify),
                            [`m-grid--${wrap}`]: _.isString(wrap),
                            [`m-grid--align-item-${align}`]: _.isString(align),
                            [`m-grid--align-content-${alignContent}`]: _.isString(alignContent)
                        },
                        ...(
                            _.isArray(className) ? className : [className]
                        )
                    )
                }>{children}</div>
            </Fragment>
        );
    }
}
