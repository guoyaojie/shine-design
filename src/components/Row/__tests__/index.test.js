import React from 'react';
// import {Enzyme} from 'tests/setup';
import {shallow} from 'enzyme';
import Row from '../index';


describe('Row', () => {
  it('是否正常渲染组件', () => {
    const wrapper = shallow(<Row>Follow</Row>);
    expect(wrapper).toMatchSnapshot();
  });
});
