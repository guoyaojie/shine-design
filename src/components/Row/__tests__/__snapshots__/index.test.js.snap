// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 是否正常渲染组件 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    align="top"
    attributes={Object {}}
    className=""
    justify="start"
    type="flex"
  >
    Follow
  </Row>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Follow",
      "className": "shine-row shine-row--root shine-row--justify-start shine-row--align-top",
    },
    "ref": null,
    "rendered": "Follow",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Follow",
        "className": "shine-row shine-row--root shine-row--justify-start shine-row--align-top",
      },
      "ref": null,
      "rendered": "Follow",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
